test/e2e/run_networking_test.go

```diff
@@ -584,6 +584,14 @@ var _ = Describe("Podman run networking", func() {
 		run := podmanTest.Podman([]string{"run", "--net=host", "--hostname", hostname, ALPINE, "hostname"})
 		run.WaitWithDefaultTimeout()
 		Expect(run.ExitCode()).To(BeZero())
-		Expect(strings.Contains(run.OutputToString(), "testctr")).To(BeTrue())
+		Expect(strings.Contains(run.OutputToString(), hostname)).To(BeTrue())
+	})
+
+	It("podman run with --net=none adds hostname to /etc/hosts", func() {
+		hostname := "testctr"
+		run := podmanTest.Podman([]string{"run", "--net=none", "--hostname", hostname, ALPINE, "hostname"})
+		run.WaitWithDefaultTimeout()
+		Expect(run.ExitCode()).To(BeZero())
+		Expect(strings.Contains(run.OutputToString(), hostname)).To(BeTrue())
```

Comment:
What exactly is this test supposed to accomplish? Actually, what exactly is this PR supposed to accomplish? As I read the description and the code -- and of course I may just be confused -- I expected the given hostname to appear in the container's `/etc/hosts` file. That is not what I'm seeing:
```console
$ ./bin/podman run --net=none --hostname abcde alpine  cat /etc/hosts
127.0.0.1       localhost localhost.localdomain
::1             localhost localhost.localdomain
```

Reply:
That sounds like a bug, because it should be in there

Reply:
So, none of the code added in this PR is ever actually invoked, because of:
https://github.com/containers/podman/blob/7774f63319be64a0a96fb7ec7c9f0b7c28faac03/libpod/container_internal_linux.go#L1243
(This is what calls `generateHosts()`, and it's not called if `netDisabled`).